### PR TITLE
fix: Accomodate partial experiment list settings

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

Changes how we use `experimentListGlobalSettingsConfig` so default values will fill in for partial settings on the exp. list.

## Test Plan

Removes repeated console.error on pages such as http://latest-main.determined.ai:8080/det/projects/1/experiments
when the settings includes:

```javascript
 "f_project-details-global": {
  "expListView": "scroll"
 },
```

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/components/UserSettings.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { Space } from 'antd';
+import { partial, type } from 'io-ts';
```
Comment (ClaireNeveu) [inline webui/react/src/components/UserSettings.tsx:2]: Idiomatic io-ts import is `import * as t from 'io-ts'`.

**Review by ClaireNeveu — CHANGES_REQUESTED** [review]

File: webui/react/src/components/UserSettings.tsx (hunk 1 above)
Comment (mapmeld) [reply webui/react/src/components/UserSettings.tsx]: ✅

File: webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts [hunk 2]
```diff
@@ -114,10 +114,10 @@ export interface F_ExperimentListGlobalSettings {
 
 const ioExpListView = union([literal('scroll'), literal('paged')]);
 
-export const experimentListGlobalSettingsConfig = type({
+export const experimentListGlobalSettingsConfig = {
   expListView: ioExpListView,
   rowHeight: ioRowHeight,
-});
+};
```
Comment (johnkim-det) [inline webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts:120]: just to make sure, using `partial` instead of `type` here doesn't work?

Comment (mapmeld) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts:120]: I had thought that using `partial()` would be necessary to un-do for `userSettings.set`, but it seems to work? That would be a simpler, smaller PR if it checks out logically.

Comment (mapmeld) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts]: ok cut down to 2-line change

**Review by ClaireNeveu — APPROVED** [review]

File: webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts (hunk 2 above)
Comment (ClaireNeveu) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts]: I think `partial` in both places is actually more correct. The Settings form doesn't actually let you unset these fields but in the application logic it's fine to unset them.

Comment (mapmeld) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts:120]: ok, rolling back this commit

Comment (ClaireNeveu) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.settings.ts:120]: Why?

**Pull request merged**